webapp-django/crashstats/symbols/views.py

```diff
@@ -67,12 +67,26 @@ def unpack_and_upload(iterator, symbols_upload, bucket_name, bucket_location):
                 "Setting %s must be set" % key
             )
 
-    conn = boto.s3.connect_to_region(
-        bucket_location,
-        aws_access_key_id=settings.AWS_ACCESS_KEY,
-        aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY,
-        calling_format=boto.s3.connection.OrdinaryCallingFormat(),
-    )
+    params = {
+        'aws_access_key_id': settings.AWS_ACCESS_KEY,
+        'aws_secret_access_key': settings.AWS_SECRET_ACCESS_KEY,
+        'calling_format': boto.s3.connection.OrdinaryCallingFormat(),
+    }
+
+    if settings.AWS_HOST and settings.AWS_PORT:
+        # NOTE(willkg): If AWS_HOST and AWS_PORT are defined, then this is a local dev environment
+        # and we need to connect to a local S3 instead of the AWS S3. In a server environment, these
+        # variables should not be set.
+        params.update({
+            'host': settings.AWS_HOST,
+            'port': settings.AWS_PORT,
+            'is_secure': settings.AWS_SECURE,
+        })
+        conn = boto.connect_s3(**params)
+
+    else:
+        conn = boto.s3.connect_to_region(bucket_location, **params)
```

Comment:
The comment here should be clear--if not, we should improve it.

This is essentially what we do in the `socorro.external.boto.connection_context` classes, too, though it's hard to see that because we have a non-trivial class hierarchy there.